sonar-orchestrator/src/main/java/com/sonar/orchestrator/Orchestrator.java

```diff
@@ -128,16 +132,29 @@ public void start() {
 
     for (String pluginKey : distribution.getLicensedPluginKeys()) {
       String license = licenses.get(pluginKey);
-      if (license != null) {
-        server.getAdminWsClient().update(new PropertyUpdateQuery(
-          licenses.licensePropertyKey(pluginKey),
-          license));
+      if (license != null && !license.isEmpty()) {
+        updateSetting(licenses.licensePropertyKey(pluginKey), license);
       }
     }
 
     buildRunner = new BuildRunner(config);
   }
 
+  private void updateSetting(String key, String value) {
+    if (getServer().version().isGreaterThanOrEquals("6.1")) {
+      newAdminWsClient().settingsService().set(SetRequest.builder().setKey(key).setValue(value).build());
```

Comment:
as discussed please use the old `server.getAdminWsClient().post()` instead of a new client. Sharing a single client becomes critical and should be addressed soon.